### PR TITLE
Add components copy of Guestbook example in C#

### PR DESCRIPTION
**Pull request by lukehoban**

### CODE REVIEW

File: kubernetes-cs-guestbook/components/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+[![Deploy](https://get.pulumi.com/new/button.svg)](https://app.pulumi.com/new)
```
Comment (mikhailshilkov) [inline kubernetes-cs-guestbook/components/README.md]: Nit: We try to use a slightly more structured template for READMEs, like [here](https://github.com/pulumi/examples/blob/master/azure-ts-aks-keda/README.md)

File: kubernetes-cs-guestbook/components/ServiceDeployment.cs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    public InputList<EnvVarArgs> Env { get; set; } = new InputList<EnvVarArgs>();
+}
+
+class ServiceDeployment : Pulumi.ComponentResource
+{
+    public Pulumi.Kubernetes.Apps.V1.Deployment Deployment;
+    public Pulumi.Kubernetes.Core.V1.Service Service;
+    public Output<string>? IpAddress;
+
+    public ServiceDeployment(string name, ServiceDeploymentArgs args, ComponentResourceOptions? opts = null) : base(name, "k8sx:serice:ServiceDeployment", opts)
```
Comment (mikhailshilkov) [inline kubernetes-cs-guestbook/components/ServiceDeployment.cs]: Nit: break into two lines

File: kubernetes-cs-guestbook/components/ServiceDeployment.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        var labels = new InputMap<string>{
```
Comment (mikhailshilkov) [inline kubernetes-cs-guestbook/components/ServiceDeployment.cs]: Nit: `{` to a new line

File: kubernetes-cs-guestbook/components/ServiceDeployment.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                    Spec = new PodSpecArgs
+                    {
+                        Containers = { container },
+                    },
+                },
+            },
+        }, new CustomResourceOptions
+        {
+            Parent = this,
+        });
```
Comment (mikhailshilkov) [inline kubernetes-cs-guestbook/components/ServiceDeployment.cs]: Nit: make it (on a new line)
`new CustomResourceOptions { Parent = this });`
Same below.

File: kubernetes-cs-guestbook/components/ServiceDeployment.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                Type = args.AllocateIPAddress.Apply(hasIp => hasIp ? (args.ServiceType ?? "LoadBalancer") : null),
+                Ports = servicePorts,
+                Selector = this.Deployment.Spec.Apply(spec => spec.Template.Metadata.Labels),
+            },
+        }, new CustomResourceOptions
+        {
+            Parent = this,
+        });
+
+        this.IpAddress = args.AllocateIPAddress.Apply(hasIp => {
```
Comment (mikhailshilkov) [inline kubernetes-cs-guestbook/components/ServiceDeployment.cs]: Nit: All `{` to new lines

**Comment by lukehoban** [comment]
> k8s parts LGTM, but this test doesn't seem to be running in CI. Not sure if that was intentional

We have not yet added the ability to run any kubernetes tests in CI for examples. Once we do (which we should do ASAP) we should add the test to the test suite.

File: kubernetes-cs-guestbook/components/README.md (hunk 1 above)
Comment (lukehoban) [reply kubernetes-cs-guestbook/components/README.md]: Good point - this was copied from the corresponding TS template - so sounds like we should update both (or open an issue to create consistency here for all existing templates).

Comment (lukehoban) [reply kubernetes-cs-guestbook/components/README.md]: Opened https://github.com/pulumi/examples/issues/493 to just to this consistently in a single pass in the near future.

**Pull request merged**